MiniSpace.Services.Email/src/MiniSpace.Services.Email.Application/Events/External/Handlers/NotificationCreatedHandler.cs

```diff
@@ -76,12 +75,22 @@ public async Task HandleAsync(NotificationCreated @event, CancellationToken canc
             await _messageBroker.PublishAsync(new EmailQueued(emailNotification.EmailNotificationId, @event.UserId));
         }
 
-
         private async Task<string> LoadHtmlTemplate(string filePath, NotificationCreated eventDetails, StudentDto student)
         {
             string htmlContent = await System.IO.File.ReadAllTextAsync(filePath);
-            htmlContent = htmlContent.Replace("{Message}", eventDetails.Message);
-            htmlContent = htmlContent.Replace("{Details}", eventDetails.Details ?? "No details provided");
+            var eventType = (NotificationEventType)Enum.Parse(typeof(NotificationEventType), eventDetails.EventType);
+            htmlContent = htmlContent.Replace("{Message}", EmailContentFactory.CreateContent(eventType, eventDetails.Details));
+
+            // Replace {Details} with empty string if eventType is TwoFactorCodeGenerated
+            if (eventType == NotificationEventType.TwoFactorCodeGenerated)
+            {
+                htmlContent = htmlContent.Replace("{Details}", string.Empty);
+            }
+            else
+            {
+                htmlContent = htmlContent.Replace("{Details}", eventDetails.Details ?? "No details provided");
+            }
+
             htmlContent = htmlContent.Replace("{CreatedAt}", eventDetails.CreatedAt.ToString("dddd, dd MMMM yyyy"));
 
             var eventTypeDescription = EmailSubjectFactory.CreateSubject(
```

Comment:
It requires refactor. Add one more factory for template context.

Reply:
Currently, we are mainly delivering Domain implementation. There're still a lot o places to conduct refactoring with dp and core patterns ussage.